internal/common/function/function.go

```diff
@@ -157,6 +167,82 @@ func ValidateRestoreSpec(ctx context.Context, clientInstance client.Client, nonA
 	return nil
 }
 
+// ValidateBslSpec return nil, if NonAdminRestore is valid; error otherwise
+func ValidateBslSpec(ctx context.Context, clientInstance client.Client, nonAdminBsl *nacv1alpha1.NonAdminBackupStorageLocation, enforcedBslSpec *velerov1.BackupStorageLocationSpec) error {
+	if nonAdminBsl.Spec.BackupStorageLocationSpec.Credential == nil {
```

Comment:
Maybe add a comment here with the issue link: https://github.com/migtools/oadp-non-admin/issues/146 as a TODO

Reply:
Fixed, will commit soon.

Reply:
Can we close this as resolved? TODO added and it's pointing 146.